test/test_script.py

```diff
@@ -943,3 +950,31 @@ def test_cmd_slice(self):
             self.parser.eval("$slice(abc; def,0)")
         with self.assertRaisesRegex(ScriptError, areg):
             self.parser.eval("$slice(abc; def),0,1,:,extra")
+
+    def test_cmd_datetime(self):
+        # Save origninal datetime object and substitute one returning
+        # a fixed now() value for testing.
+        original_datetime = datetime.datetime
+        datetime.datetime = _DateTime
+
+        context = Metadata()
+        context["foo"] = "%Y%m%d%H%M%S.%f"
+        try:
+            # Tests with context
+            self.assertScriptResultEquals("$datetime(%foo%)", "20200102123456.000789", context)
+            self.assertScriptResultEquals("$datetime($initials())", "2020-01-02 12:34:56", context)
+            # Tests with static input
+            self.assertScriptResultEquals("$datetime(\%Y\%m\%d\%H\%M\%S.\%f)", "20200102123456.000789", context)
+            # Tests with missing input
+            self.assertScriptResultEquals("$datetime()", "2020-01-02 12:34:56", context)
+            # Tests with invalid format
+            self.assertScriptResultEquals("$datetime(xxx)", "xxx", context)
+            # Tests with invalid number of arguments
+            areg = r"^Wrong number of arguments for \$datetime: Expected between 0 and 1, "
+            with self.assertRaisesRegex(ScriptError, areg):
+                self.parser.eval("$datetime(abc,def)")
```

Comment:
I don't see any test for timezone correctness in there, do I miss something?

Reply:
Nope.  I'll have to add that in.